### ISSUE TITLE
BASIC line number parsing is unreliable

### ISSUE THREAD

**Issue by benjamink** [tjboldt/ProDOS-Utilities#27]
[PR 121 in the tjboldt/Apple2-IO-RPi](https://github.com/tjboldt/Apple2-IO-RPi/pull/121) repo illustrates an issue with parsing BASIC where the line numbering gets munged causing the program to fail.

See how line `15` gets turned into `5` here: 

![a2-io-rpi-basic-list](https://user-images.githubusercontent.com/1457764/214606939-be47b4a9-5af8-4e6a-a58f-a1b2c52117da.png)

I added some extra output to the `basic.go` as follows: 

```go
                // parse line number
                if parsingLineNumber {
                        if c >= '0' && c <= '9' {
                                fmt.Println("Line num before inc: " + string(c))

                                lineNumberString += string(c)

                                fmt.Println("Line num after inc: " + string(lineNumberString))
```

When I try to `put` my BASIC file with the above logging it's possible to see the line numbering issues: 

```
...
Line num before inc: 1
Line num after inc: 1
Line num before inc: 3
Line num after inc: 13
Line num before inc: 1
Line num after inc: 1
Line num before inc: 4
Line num after inc: 14
Line num before inc: 5       # <-- seems to miss the `1` in `15`
Line num after inc: 5
Line num before inc: 1
Line num after inc: 1
Line num before inc: 6
Line num after inc: 16
Line num before inc: 1
Line num after inc: 1
Line num before inc: 7
Line num after inc: 17
...

Tried making *ALL* line numbers incremental thinking maybe jumping numbers was confusing things but even with this example the above issue happens (excuse my really bad BASIC code 😄 ):

```
10  REM *** PROMPT TO DROP TO BASIC OR OPEN BITSY BYE ***
11  GOSUB 71
12  PRINT "Choose whether to drop to BASIC or"
13  PRINT "go to the BitsyBye interface"
14  PRINT
15  PRINT "Enter BASIC? (Y/N) "
16  INPUT A$
17  IF (A$ = "Y" OR A$ = "y") THEN 50
18  PRINT CHR$ (4)"BYE"
50  REM *** DROP TO BASIC ***
51  PRINT "Use 80-column mode? (Y/N) "
52  INPUT A$
53  PRINT "Entering BASIC"
54  IF (A$ = "N" OR A$ = "n") THEN 60
55  PRINT CHR$ (4)"PR#3"
60  REM *** PRINT OUT HELP ***
61  GOSUB 71
62  PRINT "To start a shell to the RPi, type:"
63  PRINT
64  PRINT "  -shell"
65  PRINT
66  PRINT "To add RPI command to ProDOS, type:"
67  PRINT
68  PRINT "  -rpi.command"
69  PRINT
70  END
71  REM *** PRINT HEADER ***
72  HOME
73  PRINT "Apple2-IO-RPi"
74  PRINT "(c)2021-2023 Terence J. Boldt"
75  PRINT
76  PRINT "This drive is dynamically generated"
77  PRINT "from the current working directory on"
78  PRINT "the Raspberry Pi and is in RAM only."
79  PRINT
80  RETURN
```

**Comment by tjboldt** [comment]
Removing blank lines and adding a space after each token (like the word "PRINT") fixes the text file but the parser should compensate for that. I can reproduce your issue and will attempt to fix it soon.

**Comment by benjamink** [comment]
Including comment from related [PR](https://github.com/tjboldt/Apple2-IO-RPi/pull/121): 

I won't open another PR for my customized Startup.bas unless you'd like to include the extra prompts. Happy to contribute back but not sure if that's something you're looking for in this project.

Anyhow, there were a few other changes I had to make to get my BASIC to parse correctly but I finally got it after your recent updates. Few extra things I had to do:

    Change all bare PRINT lines to PRINT "" (Possibly also fixed with dummy REM after as in my next bullet - didn't try that)
    Add dummy REM statements after bare commands like HOME & END
    Remove all the asterisks (*) from my REM statements

My current working version is here: https://github.com/benjamink/Apple2-IO-RPi/blob/main/Apple2/Startup.bas
